### PR TITLE
=doc Add server-side overview page

### PR DESCRIPTION
**Pull request by jonas**
Moves the general intro from the "Server-Side Low-Level API" page to a new index page.

If this is agreed to be a good idea, we should make sure to update this PR (or the PRs <del>#1347</del> and #1351 if merged before) to list the websocket and HTTPS support pages under this page.

### CODE REVIEW

**Comment by jonas** [comment]
Rebased now that #1347 is merged

File: docs/src/main/paradox/scala/http/server-side/index.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+as it explains the underlying full-stack streaming concepts, which may be unexpected when coming
+from a background with non-"streaming first" HTTP Servers.
+@@@
+
+@@toc { depth=3 }
+
+@@@ index
+
+* [low-level-api](low-level-api.md)
+* [server-https-support](server-https-support.md)
```
Comment (raboof) [inline docs/src/main/paradox/scala/http/server-side/index.md]: This ToC is already available in the 'left menu', right?

Having an 'introduction page' makes sense to me, but not sure if it needs a ToC

Comment (jonas) [inline docs/src/main/paradox/scala/http/server-side/index.md]: `client-side/index.md` has a ToC. It shows up as the "floating boxes" which shows sub sections etc.

screen shot 2017-08-21 at 10 57 29 am

Comment (raboof) [inline docs/src/main/paradox/scala/http/server-side/index.md]: Ah it also has all subsections, that could be neat

**Comment by jonas** [comment]
Rebased to fix conflicts after the HTTP/2 docs got merged.

**Comment by jonas** [comment]
screen shot 2017-08-23 at 9 34 32 am

**Comment by raboof** [comment]
Neat improvement, but now we have "Server-Side API" and "High-level Server-Side API" next to each other in the ToC, isn't that a bit strange?

**Comment by jonas** [comment]
We can move it inside the server-side ToC or rebrand it as "Routing DSL".

**Comment by jonas** [comment]
Note there's also #436 to eventually improve the doc structure.

**Comment by jrudolph** [comment]
Yep, the TOC is known to be mess right now. We'll get to fixing that at some point...

**Comment by ktoso** [comment]
Seems like an improvement over current state so happy to merge, and we'll soon rework the TOC anyway I hope

File: docs/src/main/paradox/java/http/server-side/index.md [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../scala/http/server-side/index.md
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/server-side/index.md]: yeah seems when I pulled this symlink did not get created somehow... I wonder why it fails the other PR but this one was fine

**Comment by jonas** [comment]
In a recent PR I symlinked server-side at the directory level. I never got around to update this PR so it could be that we now have a symlink file under a symlinked directory.

**Pull request merged**